### PR TITLE
ts-migration/snapshot-processor

### PR DESCRIPTION
**Pull request by G-Rath**
Figured I might as well break this out

### CODE REVIEW

File: src/processors/snapshot-processor.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// https://eslint.org/docs/developer-guide/working-with-plugins#processors-in-plugins
+
+type PostprocessMessage = { ruleId: string };
+
+export const snapshotProcessor = {
```
Comment (SimenB) [inline src/processors/snapshot-processor.ts]: `export default`?

Comment (G-Rath) [reply src/processors/snapshot-processor.ts]: I've sort of started to [be against them](https://humanwhocodes.com/blog/2019/01/stop-using-default-exports-javascript-module/). Not strongly, but like it's a think I'm trying and have started to kind of liked it and now encourage?

But I'm not married to it, so happy to make it a default export if you'd prefer :)

Comment (SimenB) [reply src/processors/snapshot-processor.ts]: In this case I'd ether go for 2 separate named exports (sa it currently is) or a single `default` one

File: src/processors/snapshot-processor.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// https://eslint.org/docs/developer-guide/working-with-plugins#processors-in-plugins
+
+type PostprocessMessage = { ruleId: string };
```
Comment (SimenB) [inline src/processors/snapshot-processor.ts]: link to https://github.com/typescript-eslint/typescript-eslint/issues/808?

**Pull request merged**